src/rust/engine/fs/src/directory.rs

```diff
@@ -311,6 +314,43 @@ impl From<&File> for remexec::FileNode {
   }
 }
 
+#[derive(Clone, Debug, DeepSizeOf)]
+pub struct Symlink {
+  name: Name,
+  target: String,
+}
+
+impl Symlink {
+  pub fn name(&self) -> Name {
+    self.name
+  }
+
+  pub fn target(&self) -> String {
```

Comment:
Generally this should return `&str` so that callers are not forced to take ownership (and incur an allocation) if they merely want to examine the target. The caller can always call `.to_owned()` itself if it needs ownership.

Reply:
See comment about typing of `target` as a `RelativePath` or `PathBuf`.